### PR TITLE
Feat/key stage style subject listing LESQ-131

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

- Adds keystage keypad to subject listing page
- Updates useIsCurrent hook to accept keyStageSlug
- Adds transition to selected button
- Updates queries, sdk, tests and scheams files

## Issue(s)

Fixes # LESQ-131

## How to test

1. Go to https://deploy-preview-1799--oak-web-application.netlify.thenational.academy
2. Click on _______
3. You should see _______

## Screenshots

How it used to look (delete if n/a):

Screenshot 2023-08-21 at 10 51 54
Screenshot 2023-08-21 at 10 52 12

How it should now look:
Screenshot 2023-08-21 at 10 50 54
Screenshot 2023-08-21 at 10 51 15

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
Semantic HTML thing:
- They shouldn't be h3 and h4 tags as they come before the h1 tag (which means the page order is now h3, h4, h1 which is confusing). Not sure how you handle the h3 because that is a h3 on another page, but the h4s don't need to be headings on the home page... it probably won't hurt if the h3 isn't a h3 on the homepage either
- The elements should probably be put inside a list whilst we are here
- This is a fun one, gap between the edge of the blue box and viewport. Doesn't seem to matter which browser and it only happens on some devices (Huawei P30 pro on browserstack)
Screenshot_2023-08-21-12-57-10-27_40deb401b9ffe8e1df2f1cc5ba480b12

**Review by jack-skerrett-oak — APPROVED** [review]

File: src/node-lib/curriculum-api-2023/queries/subjectListing/subjectListing.gql
Comment (benprotheroe) [inline src/node-lib/curriculum-api-2023/queries/subjectListing/subjectListing.gql:1]: should this not be one mv? if we're doing one mv per page?

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/subjectListing/subjectListing.gql:1]: isn't that a scope creep for this ticket?

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/subjectListing/subjectListing.gql:1]: Ok, will bare that in mind for the future

Comment (kimon-satan) [reply src/node-lib/curriculum-api-2023/queries/subjectListing/subjectListing.gql:1]: A simple workaround would be just to cut and paste the MV and rename it. Could you do that to maintain the prinicple ?

**Review by kimon-satan — CHANGES_REQUESTED** [review]

**Review by benprotheroe — APPROVED** [review]

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**